design/anoncreds.md

```diff
@@ -0,0 +1,507 @@
+    issued before time B can be successfully verified using old key, and
+    all credentials issued between B and C becomes invalid.
+    All new credentials issued after C should be verifiable against the new key.
+    1. The Issuer needs to be able to rotate the keys multiple times. Requirement 5.ii must be true for each key rotation.  
+1. <b>Revocation</b>
+    1. Verifier needs to be able to Verify that the credential is not revoked at the current time
+    (the time when proof request is created).
+    1. Verifier needs to be able to Verify that the credential is not revoked at the given time (any time in the past).       
+1. Querying
+    1. One needs to be able to get all CRED_DEFs created by the given Issuer DID.
+    1. One needs to be able to get all SCHEMAs created by the given Issuer DID.
+    1. One needs to be able to get all SCHEMAs with the given name created by the given Issuer DID.
+    1. One needs to be able to get all SCHEMAs with the given name and version created by the given Issuer DID.
+    1. One needs to be able to get all REVOC_REG_DEFs created by the given Issuer DID.
+    1. One needs to be able to get all REVOC_REG_DEFs created by the given Issuer DID for the given CRED_DEF.    
+
+### Technical goals
+* Define how Schemas, CreDefs and RevocRegDefs are identified (seqNo, UUID, primary key tuples?)
+* Define how to deal with Requirements 6.
+* Define Revocation-related transactions and necessary changes in indy-node.
+
+
+### Referencing Schema and CredDef in Credentials and Proofs
+* Schema is referenced by unique `SchemaUUID`.
+    * Created for each new Schema.
+    * This is different from Schema Issuer DID (DID used to send `SCHEMA` txn) which can be the same for 
+    any number of Schemas.
+* CredDef is referenced by unique `CredDefUUID`.
+    * Created for each new CredDef.
+    * This is different from CredDef Issuer DID (DID used to send `CRED_DEF` txn) which can be the same for 
+    any number of CredDef.
+* RevocRegDef and RevocReg are referenced by unique `RevocRegDefUUID`.
+    * Created for each new RevocRegDef.
+    * Used to update RevocReg.
+    * This is different from RevocRegDef Issuer DID (DID used to send `REVOC_REG_DEF` txn) which can be the same for 
+    any number of RevocRegDef.
+* Each `UUID` can be considered as a global UUID within the Ledger.
+The Ledger can guarantee that malicious party can not change/break existing entity 
+defined by a UUID by 
+and checking if there is an entity with the given UUID already existent, plus
+checking the ownership when modifying existing entities (only the issuer who created en entity with the given UUID can modify it). 
+
+      
+### How Prover and Verifier get keys for Credentials and Proofs
+* Proofs and credentials come with `schemaUUID`, `credDefUUID`, `revocDefUUID`.
+* Also there can be `issuanceTime` attribute in each credential (which can be disclosed in the proof).
```

Comment:
If we want to support Requirement 6, then there must be `issuanceTime`.
It can be that this is not so critical requirement and we do not support it at all.
Or we may say that it's up to the Issuer whether to support this requirement or not, and hence `issuanceTime` may be present or not.